### DISCUSSION TITLE
URL constructor returning blank object in middleware.

### DISCUSSION THREAD

**Discussion by jperegoy** [vercel/next.js#59860] [Help]
### Summary

Running `13.4.11` with the `pages` router, simple middleware instance runs but seems to have an issue with a rudimentary JS feature, `new URL()`. Am I misunderstanding something about the namespaces or how next middleware works?

### Additional information

```js
export const config = {
  matcher: ["/((?!api|_next/static|_next/image|favicon.ico).*)", "/"],
};

export function middleware() {
  const result = new URL("https://www.vercel.com");
  console.log("result: ", result);
}
```

log output: `result: URL {  }`

I would expect to see the full object representing the aspects of the given input, e.g. 
```
{
  href: 'https://www.vercel.com/',
  origin: 'https://www.vercel.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'www.vercel.com',
  hostname: 'www.vercel.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

**Comment by rubel-mujica-wau** [comment]
I have exactly the same issue, I created a URL object somewhere and the console.log outputs the whole object, but inside the middleware prints an empty object ({}) but apparently the object is not empty or maybe has to do with some sort of "asynchrony" and the object fills in the next clock tick, because, in my case (using your example) console.log(result) outputs "URL {}" but console.log(result.href) (I guess that works with any property) outputs "https://www.vercel.com" as expected. On the other hand I use the created URL to redirect somewhere else
return NextResponse.redirect(result) and works as expected, the browser gets redirected to the url.
